homeassistant/components/homekit/sensor.py

```diff
@@ -0,0 +1,66 @@
+"""Support for homekit sensors."""
+import logging
+
+from homeassistant.core import callback
+
+from .const import DOMAIN, HOMEKIT
+from .entity import HomeKitEntity
+from .type_media_players import TelevisionMediaPlayer
+
+_LOGGER = logging.getLogger(__name__)
+
+HOMEKIT_REMOTE_MODEL = "HomeKit Remote"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the homekit sensors."""
+
+    homekit = hass.data[DOMAIN][config_entry.entry_id][HOMEKIT]
+    mac = homekit.driver.state.mac
+    entities = []
+
+    for acc in homekit.bridge.accessories.values():
+        if isinstance(acc, TelevisionMediaPlayer):
+            entities.append(RemoteKeySensor(mac, acc))
+
+    async_add_entities(entities, False)
+
+
+class RemoteKeySensor(HomeKitEntity):
```

Comment:
I don't think this is a good idea. Stateless devices should be represented as custom events.

Reply:
@MartinHjelmare Thanks for looking.

To close the loop and make sure I reconcile this properly with this feedback https://github.com/home-assistant/core/pull/35225#discussion_r419989024, a custom event is preferred here because its coming from `homekit` and not from an entity?

Reply:
There's nothing wrong with using custom events when there's a need for it. But they should not be fired from entities. If a device has states we want it to be represented by entities. If it doesn't have states, eg it's a push button, it should be represented by custom events.

Reply:
Thanks for the clarification.  I'm solving a problem that doesn't exist here so I'm going to close this and start over.